api/src/main/java/net/kyori/adventure/text/TextComponentImpl.java

```diff
@@ -49,6 +49,13 @@
     return new TextComponentImpl(Collections.emptyList(), Style.empty(), content);
   }
 
+  static @NotNull TextComponent create(final @NotNull List<? extends ComponentLike> children, final @NotNull Style style, final @NotNull String content) {
+    if (children.isEmpty() && style.isEmpty() && content.isEmpty()) {
```

Comment:
There should be a non-null check on `content` here. Also, we could check if each component in the list is empty if the list itself isn't empty? This would ensure that creating a component with empty children also returns the empty component.